site/source/docs/api_reference/emscripten.h.rst

```diff
@@ -1410,6 +1414,60 @@ Functions
 
   .. note:: This requires building with ``-s ASYNCIFY_LAZY_LOAD_CODE``.
 
+.. c:function:: emscripten_fiber emscripten_fiber_create(em_arg_callback_func func, void *arg, void *stack, int stack_size)
+
+    Creates a new fiber context. Fibers are light cooperative threads of
+    execution that can be used to implement stackful coroutines.
+
+    `func` is the fiber's entry point. It will be called with the argument `arg`
+    in the fiber's context when it's first entered.
+
+    `stack` must point to a pre-allocated memory region that will be used as the
+    fiber's stack. `stack_size` is the size of that region in bytes. `stack`
+    should be aligned to at least 16 bytes.
+
+ .. note:: If `func` returns, the entire program will end, as if `main`
+    had returned. To avoid this, you can use `emscripten_fiber_swap` to jump to
+    another fiber.
+
+.. c:function:: emscripten_fiber emscripten_fiber_create_from_current_context()
+
```

Comment:
Why not just have `emscripten_fiber_get_main`, if that is what its purposes is?

What happens if this is called from an existing fibre?   Does it still create new one?

Reply:
See my answer to the other comment below for the reason why it doesn't just return the "main" fiber.

> What happens if this is called from an existing fibre? Does it still create new one?

Yes, it creates a new one referring to the same stack. The runtime does not discriminate between the "main" and custom fibers.